src/Processors/Merges/Algorithms/MergingSortedAlgorithm.cpp

```diff
@@ -17,13 +17,15 @@ MergingSortedAlgorithm::MergingSortedAlgorithm(
     SortingQueueStrategy sorting_queue_strategy_,
     UInt64 limit_,
     WriteBuffer * out_row_sources_buf_,
+    std::optional<String> filter_column_name_,
     bool use_average_block_sizes,
     bool apply_virtual_row_conversions_)
     : header(std::move(header_))
     , merged_data(use_average_block_sizes, max_block_size_, max_block_size_bytes_)
     , description(description_)
     , limit(limit_)
     , out_row_sources_buf(out_row_sources_buf_)
+    , filter_column_position(filter_column_name_ ? header->getPositionByName(filter_column_name_.value()) : -1)
```

Comment:
Let's just in case check that column type is `UInt8`
Usually `Nullable(UInt8)` and LC/Sparce are also allowed, but here we use it only together with `RowExistsColumn::name`